arangod/V8Server/v8-user-functions.cpp

```diff
@@ -0,0 +1,153 @@
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Wilfried Goesgens
+////////////////////////////////////////////////////////////////////////////////
+
+#include "v8-user-functions.h"
+
+//#include "Basics/Result.h"
+
+#include "v8-vocbaseprivate.h"
+#include <v8.h>
+#include "V8/v8-utils.h"
+#include "V8/v8-vpack.h"
+#include "V8/v8-conv.h"
+
+#include "VocBase/Methods/AqlUserFunctions.h"
+
+using namespace arangodb;
+
+static void JS_UnregisterAQLUserFunction(v8::FunctionCallbackInfo<v8::Value> const& args) {
+  TRI_V8_TRY_CATCH_BEGIN(isolate);
+  v8::HandleScope scope(isolate);
+
+  if (args.Length() != 1) {
+    TRI_V8_THROW_EXCEPTION_USAGE("UNREGISTER_AQL_USER_FUNCTION(<name>)");
+  }
+
+  TRI_vocbase_t* vocbase = GetContextVocBase(isolate);
+
+  std::string functionName = TRI_ObjectToString(args[0]);
+
+  Result rv = unregisterUserFunction(vocbase, functionName);
+
+  if (rv.fail()) {
+    TRI_V8_THROW_EXCEPTION(rv);
+  }
+  TRI_V8_TRY_CATCH_END;
+}
+
+static void JS_UnregisterAQLUserFunctionsGroup(v8::FunctionCallbackInfo<v8::Value> const& args) {
+  TRI_V8_TRY_CATCH_BEGIN(isolate);
+  v8::HandleScope scope(isolate);
+
+  if (args.Length() != 1) {
+    TRI_V8_THROW_EXCEPTION_USAGE("UNREGISTER_AQL_USER_FUNCTION_GROUP(<group string>)");
+  }
+
+  int deleteCount;
+  TRI_vocbase_t* vocbase = GetContextVocBase(isolate);
+
+  std::string functionFilterPrefix = TRI_ObjectToString(args[0]);
+
+  Result rv = unregisterUserFunctionsGroup(vocbase, functionFilterPrefix, deleteCount);
+
+  if (rv.fail()) {
+    TRI_V8_THROW_EXCEPTION(rv);
+  }
+  TRI_V8_RETURN(v8::Number::New(isolate, deleteCount));
+  TRI_V8_TRY_CATCH_END;
+}
+
+static void JS_RegisterAqlUserFunction(v8::FunctionCallbackInfo<v8::Value> const& args) {
+  TRI_V8_TRY_CATCH_BEGIN(isolate);
+  v8::HandleScope scope(isolate);
+
+  if (args.Length() != 1) {
+    TRI_V8_THROW_EXCEPTION_USAGE("REGISTER_AQL_USER_FUNCTION(<name>, <functionbody> [, <isDeterministic>])");
+  }
+
+  TRI_normalize_V8_Obj(args, args[0]);
+
+  TRI_vocbase_t* vocbase = GetContextVocBase(isolate);
+
+  VPackBuilder builder;
+  TRI_V8ToVPackSimple(isolate, builder, args[0]);
+  bool replacedExisting;
+  Result rv = registerUserFunction(vocbase, builder.slice(), replacedExisting);
+  
+  if (rv.fail()) {
+    TRI_V8_THROW_EXCEPTION(rv);
+  }
+  TRI_V8_RETURN(replacedExisting);
+  TRI_V8_TRY_CATCH_END;
+}
+
+static void JS_GetAqlUserFunctions(v8::FunctionCallbackInfo<v8::Value> const& args) {
+  TRI_V8_TRY_CATCH_BEGIN(isolate);
+  v8::HandleScope scope(isolate);
+
+  if (args.Length() > 1) {
+    TRI_V8_THROW_EXCEPTION_USAGE("GET_AQL_USER_FUNCTIONS([<group-filter-string>])");
+  }
+  TRI_vocbase_t* vocbase = GetContextVocBase(isolate);
+  
+  std::shared_ptr<VPackBuilder> result;
+
+  result.reset(new VPackBuilder);
```

Comment:
fuse this line and the former one into an `auto result = std::make_shared<VPackBulder>();`